util/sim/data_utils.py

```diff
@@ -6,15 +6,47 @@
 
 import struct
 from datetime import datetime
+import torch
+import numpy as np
 
 
 def emit_license():
     s = (f"// Copyright {datetime.now().year} ETH Zurich and University of Bologna.\n"
          f"// Licensed under the Apache License, Version 2.0, see LICENSE for details.\n"
-         f"// SPDX-License-Identifier: Apache-2.0\n\n")
+         f"// SPDX-License-Identifier: Apache-2.0\n")
     return s
 
 
+def floating_point_torch_type(precision):
+    prec_to_torch_type_map = {
+        '64': torch.float64,
+        '32': torch.float32,
+        '16': torch.float16,
+        '8': None
+    }
+    return prec_to_torch_type_map[precision]
```

Comment:
There is no need to wrap the dict as a function. We could declare `TORCH_DTYPE`, `NP_DTYPE` etc. as dicts here and import it in all the verify scripts

Reply:
@colluca

Reply:
True, I just prefer functions in general for extensibility, we could e.g. add type-checking to it.